frontend/src/components/common/Table.tsx

```diff
@@ -20,39 +21,59 @@ interface TableProps {
 }
 
 export const Table = ({ headers, rows }: TableProps): React.ReactElement => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 8; // can edit this to show how many itemsperpage we want
+  const totalItems = rows.length;
+  const totalPages = Math.ceil(totalItems / itemsPerPage);
+
+  // Calculate the index range of items to display on the current page
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const endIndex = Math.min(startIndex + itemsPerPage, totalItems);
+  const itemsToShow = rows.slice(startIndex, endIndex);
+
+  // Define the onPageChange function to update the currentPage state
+
   return (
-    <TableContainer
-      padding="0.5em"
-      border="1px solid"
-      borderColor="#E2E8F0"
-      borderRadius="12px"
-      minWidth="100%"
-    >
-      <T>
-        <Thead>
-          <Tr _hover={{ pointerEvents: "none" }}>
-            {headers.map((header, index) => (
-              <Th key={index}>{header}</Th>
-            ))}
-          </Tr>
-        </Thead>
-        <Tbody>
-          {rows.map((row, rowIndex) => (
-            <Tr
-              key={rowIndex}
-              backgroundColor={rowIndex % 2 === 0 ? "blue.50" : "grey.50"}
-            >
-              {row.values.map((value, cellIndex) => (
-                <Td key={value} fontWeight={cellIndex === 0 ? "bold" : ""}>
-                  {value}
-                </Td>
+    <>
+      <TableContainer
+        padding="0.5em"
+        border="1px solid"
+        borderColor="#E8EDF1"
+        borderRadius="12px"
+        minWidth="100%"
+      >
+        <T>
+          <Thead>
+            <Tr _hover={{ pointerEvents: "none" }}>
+              {headers.map((header, index) => (
+                <Th key={index}>{header} </Th>
               ))}
-              <Td width="5%">{row.menu}</Td>
             </Tr>
-          ))}
-        </Tbody>
-      </T>
-    </TableContainer>
+          </Thead>
+          <Tbody>
+            {itemsToShow.map((row, rowIndex) => (
+              <Tr
+                key={rowIndex}
+                backgroundColor={rowIndex % 2 === 0 ? "#E8EDF1" : "#FFFFFF"}
+              >
+                {row.values.map((value, cellIndex) => (
+                  <Td key={value} fontWeight={cellIndex === 0 ? "500" : ""}>
```

Comment:
```suggestion
                  <Td key={value} fontWeight={cellIndex === 0 ? "bold" : ""}>
```
try not to change existing code!